source/read-write-pref.txt

```diff
@@ -0,0 +1,284 @@
+You can also control how the library waits for acknowledgment of read and write operations
+on a replica set by setting a read and write concern.
+
+This section shows how to configure the read preference, read concern, and write
+concern at various levels by passing an options array parameter to the following
+methods:
+
+- :ref:`MongoDB\Client::__construct() <php-read-write-client>`: Configures client-level
+  settings
+- :ref:`MongoDB\Client::startSession() <php-read-write-session>`: Configures session-level
+  settings
+- :ref:`MongoDB\Driver\Session::startTransaction() <php-read-write-transaction>`:
+  Configures transaction-level settings
+- :ref:`MongoDB\Client::selectDatabase() <php-read-write-database>`: Configures
+  database-level settings
+- :ref:`MongoDB\Client::selectCollection() <php-read-write-collection>`: Configures read
+  collection-level settings
+
+.. _php-read-write-client:
+
+Client Configuration
+~~~~~~~~~~~~~~~~~~~~
+
+This example shows how to set the read preference, read concern, and
+write concern of a ``MongoDB\Client`` instance by passing an array to
+the constructor. The code configures the following settings:
+
+- ``secondary`` read preference: Read operations retrieve data from 
+  secondary replica set members
+- ``local`` read concern: Read operations return the instance's most recent data 
+  without guaranteeing that the data has been written to a majority of the replica
+  set members
+- ``2`` write concern: The primary and one secondary replica set member
+  must acknowledge the write operation
+
+.. literalinclude:: /includes/read-write-pref.php
+    :language: php
+    :dedent:
+    :start-after: start-client-settings
+    :end-before: end-client-settings
+
+Alternatively, you can specify the read and write settings in the connection
+URI, which is passed as a parameter to the ``MongoDB\Client`` constructor:
+
+.. literalinclude:: /includes/read-write-pref.php
+    :language: php
+    :dedent:
+    :start-after: start-client-settings-uri
+    :end-before: end-client-settings-uri
+
+.. _php-read-write-session:
+
+Session Configuration
+~~~~~~~~~~~~~~~~~~~~~
+
+This example shows how to set the read preference, read concern, and
+write concern of a session by passing an array to the ``startSession()``
+method. The code configures the following settings:
+
+- ``PRIMARY_PREFERRED`` read preference: Read operations retrieve data from 
+  the primary replica set member, or secondary members if the primary is unavailable
+- ``LOCAL`` read concern: Read operations return the instance's most recent data 
+  without guaranteeing that the data has been written to a majority of the replica
+  set members
+- ``MAJORITY`` write concern: The majority of all replica set members
+  must acknowledge the write operation
+
+
+.. literalinclude:: /includes/read-write-pref.php
+    :language: php
+    :dedent:
+    :start-after: start-session-settings
+    :end-before: end-session-settings
+
```

Comment:
S: somewhere in this section you can note that you must instantiate the appropriate type to set the options (compared to client which took strings)

Reply:
Or perhaps that info can go in client section and just inform that the syntax is different only for client